eth-connector/src/lib.rs

```diff
@@ -498,8 +526,14 @@ impl Withdraw for EthConnectorContract {
         let sender_id = env::predecessor_account_id();
         // Burn tokens to recipient
         self.ft.internal_withdraw(&sender_id, amount);
+
+        let fee_amount =
+            self.calculate_fee_amount(amount.into(), FeeType::Withdraw, Some(sender_id));
+        // Mint fee
+        self.mint_eth_on_near(&env::current_account_id(), fee_amount.0);
+
         WithdrawResult {
-            amount,
+            amount: amount.checked_sub(fee_amount.0).sdk_unwrap(),
```

Comment:
It is dangerous to panic because other components/contracts that may call this method don't expect any panics and don't implement a refund functionality, so it was implemented as described in the specification:

If the the calculated fee_amount is higher than the withdraw/deposit amount then charge the full amount as a fee and don't transfer anything to the recipient.

Reply:
10d1ab3ac41e26e55b890e87629fee46333d37aa